src/EmbeddedVsix/EmbeddedVsix.csproj

```diff
@@ -0,0 +1,115 @@
+    <CopyBuildOutputToOutputDirectory>false</CopyBuildOutputToOutputDirectory>
+    <CopyOutputSymbolsToOutputDirectory>false</CopyOutputSymbolsToOutputDirectory>
+    <StartAction>Program</StartAction>
+    <StartProgram Condition="'$(DevEnvDir)' != ''">$(DevEnvDir)devenv.exe</StartProgram>
+    <StartArguments>/rootsuffix Exp</StartArguments>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
+    <DebugSymbols>true</DebugSymbols>
+    <DebugType>full</DebugType>
+    <Optimize>false</Optimize>
+    <OutputPath>bin\Debug\</OutputPath>
+    <DefineConstants>DEBUG;TRACE</DefineConstants>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
+    <DebugType>pdbonly</DebugType>
+    <Optimize>true</Optimize>
+    <OutputPath>bin\Release\</OutputPath>
+    <DefineConstants>TRACE</DefineConstants>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="Microsoft.VSSDK.BuildTools" Version="16.8.3038">
+      <IncludeAssets>runtime; build; native; contentfiles; buildtransitive</IncludeAssets>
+      <PrivateAssets>all</PrivateAssets>
+    </PackageReference>
+    <None Include="source.extension.vsixmanifest">
+      <SubType>Designer</SubType>
+    </None>
+    <Content Include="ReadMe.txt" />
+  </ItemGroup>
+  <Import Project="..\EmbeddedSonarAnalyzer.props" />
+  <!-- ********************************************************** -->
+  <!-- SonarJS plugin handling -->
+  <!-- Download and unpack the jar, select the required files and -->
+  <!-- add them to the VSIX.                                      -->
+  <!-- ********************************************************** -->
+  <PropertyGroup>
+    <!-- Download url for the SonarJS plugin. Update this url to embed a new version of the plugin. -->
+    <PluginUrl>https://binaries.sonarsource.com/Distribution/sonar-javascript-plugin/sonar-javascript-plugin-$(EmbeddedSonarJSAnalyzerVersion).jar</PluginUrl>
+    <TypeScriptFolderName>ts</TypeScriptFolderName>
+    
+    <!-- Folder the embedded files should be copied to. This folder should be excluded from source code control. -->
+    <TypeScriptTargetDirectory>$(MSBuildThisFileDirectory)$(TypeScriptFolderName)</TypeScriptTargetDirectory>
+    
+    <!-- Hook the target that fetches the additional files into to the build process. 
+         This needs to be set after the SDK targets have been imported.
+    -->
+    <PrepareForBuildDependsOn>$(PrepareForBuildDependsOn);GetTypeScriptFiles</PrepareForBuildDependsOn>
+    <!-- By default the SDK targets only zip Release builds. We'll always zip since the file is very large otherwise. -->
+    <ZipPackageCompressionLevel>Normal</ZipPackageCompressionLevel>
+  </PropertyGroup>
+  
+  <ItemGroup Label="GetTypScriptFiles inputs and outputs">
```

Comment:
This ItemGroup is new. It doesn't completely stop the VSIX from being rebuilt; for some reason MSBuild always seems to think that some manifest-related cache file has changed so the VSIX still needs to be rebuilt.

I'll look at that separately, along with making matching changes to the CFamily fetching code.